Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Search.hs

```diff
@@ -122,9 +123,23 @@ handler merchantId sReq = do
       toLocationLatLong = sReq.dropLocation
   unlessM (rideServiceable org.geofencingConfig QGeometry.someGeometriesContain fromLocationLatLong (Just toLocationLatLong)) $
     throwError RideNotServiceable
-  result <- getDistanceAndDuration merchantId fromLocationLatLong toLocationLatLong sReq.routeDistance sReq.routeDuration
-  CD.cacheDistance sReq.transactionId (result.distance, result.duration)
   Redis.setExp (CD.deviceKey sReq.transactionId) sReq.device 120
+  result <-
+    case sReq.parentSearchId of
+      Just searchId -> do
+        mbDistanceAndDuration <- CD.getCacheDistance searchId
+        case mbDistanceAndDuration of
+          Just distanceAndDuration -> do
+            CD.cacheDistance sReq.transactionId distanceAndDuration
+            return DistanceAndDuration {distance = fst distanceAndDuration, duration = snd distanceAndDuration}
+          Nothing -> do
+            distanceAndDuration <- getDistanceAndDuration merchantId fromLocationLatLong toLocationLatLong sReq.routeDistance sReq.routeDuration
+            CD.cacheDistance sReq.transactionId (distanceAndDuration.distance, distanceAndDuration.duration)
+            return distanceAndDuration
+      Nothing -> do
+        distanceAndDuration <- getDistanceAndDuration merchantId fromLocationLatLong toLocationLatLong sReq.routeDistance sReq.routeDuration
+        CD.cacheDistance sReq.transactionId (distanceAndDuration.distance, distanceAndDuration.duration)
```

Comment:
same here.

Reply:
if parentSearchId is Nothing it means it is first time search then we need to store the things